server/middleware/OktaServerSideAuthMiddleware.ts

```diff
@@ -0,0 +1,90 @@
+import { joinUrl } from '@guardian/libs';
+import type { NextFunction, Request, Response } from 'express';
+import fetch from 'node-fetch';
+import {
+	clearIdentityLocalState,
+	getIdentityLocalState,
+	setIdentityLocalState,
+} from '@/server/IdentityLocalState';
+import { OAuthAccessTokenCookieName } from '@/server/oauth';
+import { getConfig as getOktaConfig } from '@/server/oktaConfig';
+import { requiresSignin } from '../../shared/requiresSignin';
+
+declare const CYPRESS: string;
+
+type OktaUserInfo = {
+	legacy_identity_id: string;
+	email: string;
+	name: string;
+};
+
+export const withOktaSeverSideValidation = async (
+	req: Request,
+	res: Response,
+	next: NextFunction,
+) => {
+	if (CYPRESS === 'SKIP_IDAPI') {
+		return next();
+	}
+	const oktaConfig = await getOktaConfig();
+	if (!oktaConfig.useOkta) {
+		console.log('Okta disabled, skipping Okta server side validation...');
+		return next();
+	}
+
+	console.log('Validating token server side ...');
+	const signinRequired = () => requiresSignin(req.originalUrl);
+
+	const locallyValidatedIdentityData = getIdentityLocalState(res);
+	const accessToken = req.signedCookies[OAuthAccessTokenCookieName];
+	if (!accessToken || !locallyValidatedIdentityData?.userId) {
+		if (signinRequired()) {
+			console.error(
+				'error: no access token or user in request for a sign-in required endpoint! this should have failed local validation',
+			);
```

Comment:
I tried to format this as 
```typescript
		console.error(	'error: no access token or user in request for a sign-in required endpoint! this should have failed local validation');
```
but there's some autoformatting when I commit that really wants me to put in it in multiple lines and with a comma and an empty second argument

Reply:
That's fine, it just keeps lines to under 80 or 100 characters or something.

Reply:
That's just the ESLint setup here, I guess.